pandas/core/tools/datetimes.py

```diff
@@ -50,11 +50,12 @@ def _maybe_cache(arg, format, cache, convert_listlike):
     from pandas import Series
     cache_array = Series()
     if cache:
-        # Perform a quicker unique check
         from pandas import Index
-        if not Index(arg).is_unique:
-            unique_dates = algorithms.unique(arg)
-            cache_dates = convert_listlike(unique_dates, True, format)
+        # GH26078
+        unique_dates = Index(arg).unique()
+        if len(unique_dates) < len(arg):
+            cache_dates = convert_listlike(unique_dates.to_numpy(),
```

Comment:
why did you add the ``.to_numpy()``?

remove the GH issue reference, instead restore the comment

Reply:
Before our changes `algorithms.unique(arg)` always returns `numpy.ndarray` when calling from `_maybe_cache`, so we use this cast to not change the previous behavior.